File: 4100-complete-prime-number/complete-prime-number.js
/**
 * @param {number} num
 * @return {boolean}
 */
var completePrime = function(num) {
    function isPrime(n){
        if(2>n) return false;
        var i=2;
        while(i*i<=n){
            if(n%i==0){
                return false;
            }
            i++;
        }
        return true;
    }
    let s=num.toString();
    let n=s.length;
    for(var i=0;i<n;i++){
    let left=parseInt(s.substring(0,i+1));
    let right=parseInt(s.substring(n-i-1));
    if(!isPrime(left) || !isPrime(right)){
        return false;
    }
    }
    return true;
};

File: 4100-complete-prime-number/complete-prime-number.java
class Solution {
    public boolean isPrime(int n){
        if(n<2) return false;
        for(int i=2;i*i<=n;i++){
            if(n%i==0){
                return false;
            }
        }
        return true;
    }
    public boolean completePrime(int num) {
        String str=Integer.toString(num);
        for(int i=0;i<str.length();i++){
            if(!isPrime(Integer.parseInt((str.substring(0,i+1)))) || !isPrime(Integer.parseInt((str.substring(str.length()-1-i,str.length()))))){
                return false;
            }
        }
        return true;
    }
}